### PR TITLE
London class 7- Nataline Green-Git Session-Week 7

### PR DESCRIPTION
**Pull request by NatalineGn**
# Your Details

- Your Name: Nataline Green
- Your City: London
- Your Slack Name: Nat G

# Homework Details

- Module: Git Session
- Week: 7

### CODE REVIEW

**Comment by DibaKarimi** [comment]
Hi Nataline, Well done. Great job!
I wasn't sure about the number of week ,I looked up on pull 1 , Chris mentioned as week 1.

**Comment by AbdulazizAlhasan** [comment]
Good work Nataline.

**Comment by AhmadJanah** [comment]
Hi Nataline, It is a great job, well done.

Your answer to question 11 is great but if you add that it will open a pull request.

Thanks.

File: homework.md [hunk 1]
```diff
@@ -44,37 +44,39 @@ Using CSS, how would I select every `<h1>` tag on the page?
...
+A scripting or programming language that allows you to implement complex features on web pages 
 
 ## Question 8
 
 What three problem does Git & Github solve?
 
-When people want to show off code to each other they can put it on Github
+1.Git is a version control system that lets you manage and keep track of your source code history. 
+2.GitHub is a cloud-based hosting service that lets you manage Git repositories. 
+3.If you have open-source projects that use Git, then GitHub is designed to help you better manage them.
```
Comment (louisechow) [inline homework.md:56]: You've got a great list here of the features that Git and Github provide. 👍

Can you describe what problems these features solve? What problem does a cloud-based hosting service solve? What management problem does Git solve?

File: homework.md [hunk 2]
```diff
@@ -44,37 +44,39 @@ Using CSS, how would I select every `<h1>` tag on the page?
...
 
 ## Question 9
 
 What happens when you `fork` a repository?
 
-You delete it
+You create a personal copy of someone else's project, this allows you to later submit your contribution before affecting changes to the original project.
```
Comment (louisechow) [inline homework.md:62]: Great answer! 👍
One other thing to add is where this copy resides?

File: homework.md [hunk 3]
```diff
@@ -44,37 +44,39 @@ Using CSS, how would I select every `<h1>` tag on the page?
...
 
-It send it to a friend
+it is copied onto your local computer
 
 ## Question 11
 
 What is a Pull Request?
 
-When you send a file over the internet
+When you open a pull request, you're proposing your changes and requesting that someone review and pull in your contribution and merge them into their branch
```
Comment (louisechow) [inline homework.md:73]: There can also be a case where you are the owner of the repository and you are asking your team members to review your code. In this case, you may be the person responsible for the merge not the person who reviews the code.

File: homework.md [hunk 4]
```diff
@@ -86,19 +88,19 @@ You send the changes of your local repository to Github
...
 
-A place for us to store and share our code.
+A collaborative place for us to store and share our code.
 
 ## Question 15
 
 What does it mean to "Commit your changes"?
 
-Send you code to Github
+Save your changes in Github
```
Comment (louisechow) [inline homework.md:98]: Github is the place where the remote repositories are stored. Does `commit` save changes to remote repositories?

**Comment by louisechow** [comment]
Great job with the homework Nataline 🌟

I've added some extra questions and comments on some of your answers. See if you are able to answer those questions and feel free to reach out if you want to discuss any of the topics more. 🙂

**Pull request closed without merging**